dotnet/src/SemanticKernel.Abstractions/Diagnostics/HttpResponseMessageExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Diagnostics;
+using System.Net.Http;
+using Microsoft.Extensions.Logging;
+
+/// <summary>
+/// Provides extension methods for <see cref="HttpResponseMessage"/>.
+/// </summary>
+public static class HttpResponseMessageExtensions
+{
+    /// <summary>
+    /// Ensures that the HTTP response was successful; otherwise, throws an <see cref="HttpOperationException"/>.
+    /// </summary>
+    /// <param name="response">The <see cref="HttpResponseMessage"/> to check for success.</param>
+    /// <param name="responseContent">Optional. The content of the HTTP response.</param>
+    /// <param name="logger">Optional. The logger to use for logging errors.</param>
+    public static void EnsureSuccess(this HttpResponseMessage response, string? responseContent = null, ILogger? logger = null)
```

Comment:
This doesn't seem like quite the right pattern...
Get the response, then get the contents, then ensure success...

Since you have the `HttpResponseMessage` object here already, and the only time this method reads `responseContent`, can you just remove this arg and replace the line in the "catch" below with something like:

var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
throw new HttpOperationException(response.StatusCode, responseContent, e.Message, e)

Reply:
Yeah, that's actually how I wanted and tried to implement it initially, but then I decided not to proceed with that approach. The reason is that the content reading operation is asynchronous and requires awaiting, which means the `EnsureSuccess` method will have to be marked as `async`, and as a result, its call pattern will be like this: `await response.EnsureSuccess().ConfigureAwait(false)` instead of the simple `response.EnsureSuccess()`. So, if we are okay with the async nature of the new `EnsureSuccess` extension method that needs to be awaited and configured with the `ConfigureAwait` call, I can make that change.

Reply:
I see the problem, and why that might be a bit clumsy. A few ideas...

**Sync-over-async within EnsureSuccess**: since you're only reading the stream in the throw condition on EnsureSuccess(), the calling code will never actually be awaiting anything (it's either a sync response or it's throwing). This is one of the rare cases where awaiting synchronously is actually okay.

**EnsureSuccess takes a task**: For example:
`Task<HttpResponseMessage> EnsureSuccessAsync(this Task<HttpResponseMessage> responseTask)`
This way, you can call like 'var result = await httpClient.SendAsync(request).EnsureSuccess().ConfigureAwait(false);`

**Extensions on httpClient**: like `httpClient.SendAndEnsureSuccessAsync()'

Reply:
I'm not clear on why this is necessary at all. The purpose here is to translate the HttpRequestException into an HttpOperationException?  What about when an HttpRequestException is thrown from the actual SendAsync or while reading the response content... what handles translating those?